crypto/http/http_client.c

```diff
@@ -275,42 +278,36 @@ static int set1_content(OSSL_HTTP_REQ_CTX *rctx,
             && !OSSL_HTTP_REQ_CTX_add1_header(rctx, "Connection", "keep-alive"))
         return 0;
 
+    BIO_free(rctx->req);
+    rctx->req = NULL;
     if (req == NULL)
         return 1;
     if (!rctx->method_POST) {
         ERR_raise(ERR_LIB_HTTP, ERR_R_SHOULD_NOT_HAVE_BEEN_CALLED);
         return 0;
     }
+    if (!BIO_up_ref(req))
+        return 0;
+    rctx->req = req;
```

Comment:
There are still failures further down where the req would be assigned but set1_content would return 0. I do not like that.

Reply:
It would not hurt because `rctx->req` would not be used in case of error and `OSSL_HTTP_REQ_CTX_free()` will free it anyway, but okay, reverted that.